hack/licence-detector/whitelist/whitelist.go

```diff
@@ -0,0 +1,53 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package whitelist
+
+import (
+	"fmt"
+
+	"github.com/elastic/cloud-on-k8s/hack/licence-detector/dependency"
+)
+
+// CheckWhitelist returns an error if any dependencies use a licence that is not permitted by Elastic
+func CheckWhitelist(deps *dependency.List) error {
+	err := "Dependency %s uses licence %s which is not whitelisted"
+	for _, dep := range deps.Direct {
+		if !(inWhitelist(dep.LicenceType)) {
+			return fmt.Errorf(err, dep.Name, dep.LicenceType)
+		}
+	}
+	for _, dep := range deps.Indirect {
+		if !(inWhitelist(dep.LicenceType)) {
+			return fmt.Errorf(err, dep.Name, dep.LicenceType)
+		}
+	}
+	return nil
+}
+
+func inWhitelist(depName string) bool {
+	// this is not an exhaustive list of Elastic-approved licences, but includes all the ones we use to date
+	whitelist := []string{
```

Comment:
Maybe pass this array to this function from `CheckWhitelist` so that it doesn't need to be initialized for every dependency?

Reply:
It didn't occur to me before: what if we simply make this a `map[string]struct{}`? Then we don't need the `inSlice` function and the lookup should be O(1) instead of O(n).